server/util/src/gs1.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::HashMap;
+
+use chrono::NaiveDate;
+
+#[derive(Debug)]
+pub enum GS1ParseError {
+    InvalidFormat,
+}
+
+#[derive(Debug)]
+pub struct GS1 {
+    gs1: HashMap<String, String>,
+}
+
+impl GS1 {
+    pub fn new() -> Self {
+        Self {
+            gs1: HashMap::new(),
+        }
+    }
+
+    pub fn parse(gs1_input: String) -> Result<Self, GS1ParseError> {
+        let gs1 = parse_gs1_string(gs1_input)?;
+
+        Ok(Self { gs1 })
+    }
+
+    pub fn get(&self, key: &str) -> Option<&String> {
+        self.gs1.get(key)
+    }
+
+    pub fn gtin(&self) -> Option<String> {
+        self.gs1.get("01").cloned()
```

Comment:
Might be nice to add a comment with link to spec for where you got these from? Though if this is all about to change then maybe fine 😁

Reply:
It's all changing...